conn_pool.go

```diff
@@ -335,8 +398,16 @@ func (p *ConnPool) Prepare(name, sql string) (*PreparedStatement, error) {
 	if err != nil {
 		return nil, err
 	}
-	ps, err := c.Prepare(name, sql)
 	p.availableConnections = append(p.availableConnections, c)
+
```

Comment:
This is how it used to be.
Anyway we should not be signalling until we exit from Prepare.